### DISCUSSION TITLE
Remove black shadows around the main shape in transparent image

### DISCUSSION THREAD

**Discussion by flexeo** [ImageMagick/ImageMagick#4579] [Help]
I'm looking for an ImageMagick command to make a post processing on a background removal application.

I can't remove the black shadows around the main shape in pictures.

I tried with the Erode option but the result is not very satisfying:
`convert input.png -channel A  -morphology Erode Disk:3 +channel output.png`

Do you have an idea to do that with ImageMagick? 

_**With transparent tile:**_

![1](https://user-images.githubusercontent.com/22362051/145684279-8214c863-922b-4e27-8f84-a0098bef60f7.png)
![2](https://user-images.githubusercontent.com/22362051/145684287-00d11015-90d3-4c81-a969-267575a3991b.png)
![3](https://user-images.githubusercontent.com/22362051/145684290-40be3bc5-45cc-4163-b746-3efd1bead832.png)
![4](https://user-images.githubusercontent.com/22362051/145684296-12e94180-1b49-4ef8-be7d-7985cf817b92.png)

**_Original:_**
![1](https://user-images.githubusercontent.com/22362051/145684556-600926c2-990c-4ff6-b23f-c6c12186f5d6.png)
![2](https://user-images.githubusercontent.com/22362051/145684562-0dce883d-11f2-43e3-ad98-7c52dcdd8fca.png)
![3](https://user-images.githubusercontent.com/22362051/145684637-4435da71-e2a0-4fcc-9b67-a6dce46c370a.png)
![4](https://user-images.githubusercontent.com/22362051/145684570-8d3884b3-57e7-4e29-88c5-4dcdb8a55ca4.png)

**Comment by fmw42** [comment]
Here is the way to do that. Extract the alpha channel. Threshold it and blur it then apply level to it. Adjust the threshold blur and level values as desired to remove the shadow.  Here are two possible adjustments. Unix syntax for IM 6.

```
convert 145684556-600926c2-990c-4ff6-b23f-c6c12186f5d6.png \
\( +clone -alpha extract -threshold 75% -blur 0x4 -level 50x100% \) \
-alpha off -compose copy_opacity -composite \
result1.png
```

![result1](https://user-images.githubusercontent.com/13458218/145685317-53aa2281-9c33-4b33-b202-d08e405569ff.png)



```
convert 145684556-600926c2-990c-4ff6-b23f-c6c12186f5d6.png \
\( +clone -alpha extract -threshold 50% -blur 0x4 -level 75x100% \) \
-alpha off -compose copy_opacity -composite \
result2.png

```
![result2](https://user-images.githubusercontent.com/13458218/145685323-7c6c74cc-f4f0-408c-8a09-51369b154a1d.png)

**Comment by snibgo** [comment]
The transparencies are weird: images do not seem to have any fully opaque pixels. Is that deliberate?

The gray pixels around the figures have low transparency. So one method is to adjust the transparency levels, like this:
```
magick 145684556-600926c2-990c-4ff6-b23f-c6c12186f5d6.png -channel A -level 90,100% +channel x.png
```
